ui/src/router.js

```diff
@@ -12,7 +12,18 @@ let router = new Router({
       path: "/",
       name: "home",
       props: true,
-      component: () => import(/* webpackChunkName: "home" */ "./views/Home.vue")
+      component: () => import(/* webpackChunkName: "home" */ "./views/Home.vue"),
+      beforeEnter (to, from, next) {
+        var roles=store.state.roles;
+        if(roles == "Admin" || roles == "Edit" || roles == "View")
```

Comment:
I can see this type of check being a bit unwieldy as we add more and more roles. These conditionals can get out of hand. I can see two different ways of solving this:

1) When a role is added to a user, it automatically adds the roles under it. So if I add admin to a user, then edit and view are also automatically added. This would require an update to the edit role page so you could only edit the higher level role.

2) Create a method in the frontend called hasRole and let the logic live there. So this line would be simplified to:
--- hasRole("view")

and the method would look to see what role this user has and return true / false based on that role. So if a user had the admin role, it would return true because we would assume the view role. I think this one is probably less work.

Reply:
I have opted for the option 2. And created a fonction to check if the role is Allowed to access the page. But it does not return true/false since it update the cookies variable. And for each page, I have created an array of roles allowed to access the correspondant page